sdk/storage/azure-storage-common/src/test/java/com/azure/storage/common/policy/StorageBearerTokenChallengeAuthorizationPolicyTests.java

```diff
@@ -3,209 +3,262 @@
-        assertEquals("https://storage.azure.com", scope2);
-        assertEquals("https://login.microsoftonline.com/" + tenantId2 + "/oauth2/authorize", authorization2);
+        assertNotNull(result);
+        assertEquals(1, result.getScopes().size());
+        assertEquals("https://storage.azure.com/.default", result.getScopes().get(0));
+        assertNull(result.getTenantId());
+        assertTrue(result.isCaeEnabled());
     }
 
     @Test
-    public void testExtractTenantIdFromUri() {
-        StorageBearerTokenChallengeAuthorizationPolicy policy
-            = new StorageBearerTokenChallengeAuthorizationPolicy(mockCredential, "https://storage.azure.com/.default");
+    public void testCreateTokenRequestContextWithAuthUrlOnly() {
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put("authorization_uri", "https://login.microsoftonline.com/tenant/oauth2/authorize");
 
-        String uri = "https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/oauth2/authorize";
-        String expectedTenantId = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+        TokenRequestContext result = policy.createTokenRequestContext(attributes);
 
-        String actualTenantId = policy.extractTenantIdFromUri(uri);
-
-        assertEquals(expectedTenantId, actualTenantId);
+        assertNotNull(result);
+        assertEquals(1, result.getScopes().size());
+        assertEquals("https://storage.azure.com/.default", result.getScopes().get(0));
+        assertEquals("tenant", result.getTenantId());
+        assertTrue(result.isCaeEnabled());
     }
 
     @Test
-    public void testExtractTenantIdFromUriInvalidUri() {
-        StorageBearerTokenChallengeAuthorizationPolicy policy
-            = new StorageBearerTokenChallengeAuthorizationPolicy(mockCredential, "https://storage.azure.com/.default");
+    public void testAuthorizeRequestOnChallengeWithValidChallenge() {
+        // Create a mock response with WWW-Authenticate header
+        HttpRequest request = new HttpRequest(HttpMethod.GET, "https://storage.azure.com");
+        HttpHeaders headers = new HttpHeaders().set(HttpHeaderName.WWW_AUTHENTICATE,
+            "Bearer resource_id=\"https://storage.azure.com\", authorization_uri=\"https://login.microsoftonline.com/tenant/oauth2/authorize\"");
+        HttpResponse response = new MockHttpResponse(request, 401, headers);
+
+        // Create a mock pipeline call context
+        HttpPipelineCallContext context = createMockCallContext(request);
+
+        // Test that the policy correctly processes the challenge and returns true
+        boolean result = policy.authorizeRequestOnChallengeSync(context, response);
+
+        assertTrue(result);
+        assertNotNull(request.getHeaders().getValue(HttpHeaderName.AUTHORIZATION));
+        assertTrue(request.getHeaders().getValue(HttpHeaderName.AUTHORIZATION).startsWith("Bearer "));
```

Comment:
Would it be possible to assert aspects such as the tenant id being set or the scopes being updated? Mainly that would add nice end to end coverage from the perspective of the policy class that it is functionally behaving as expected.